src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -49,94 +41,18 @@ public static function onShowSearchHitTitle(
 		array &$query,
 		array &$attributes
 	): void {
-		$itemId = self::getItemId( $title );
+		$itemId = WikibaseFacetedSearchExtension::getInstance()->getPageItemLookup()->getItemId( $result->getTitle() );
 
 		if ( $itemId === null ) {
 			return;
 		}
 
-		$pageTitle = self::getItemPage( $itemId );
+		$rewrittenLink = WikibaseFacetedSearchExtension::getInstance()
+			->newPageSearchResultRewriter( $specialSearch->getContext() )
+			->getRewrittenLink( $itemId );
 
-		if ( $pageTitle === null ) {
-			return;
-		}
-
-		$title = $pageTitle;
-
-		if ( !( $result instanceof EntityResult ) ) {
-			self::rewriteLinkForNonEntityResult(
-				self::newLabelDescriptionLookup( $specialSearch->getContext() ),
-				self::newLinkFormatter( $specialSearch->getLanguage() ),
-				$itemId,
-				$titleSnippet,
-				$attributes
-			);
-		}
-	}
-
-	private static function getItemId( Title $title ): ?ItemId {
-		$entityId = WikibaseRepo::getEntityIdLookup()->getEntityIdForTitle( $title );
-
-		if ( $entityId instanceof ItemId ) {
-			return $entityId;
-		}
-
-		return null;
-	}
-
-	private static function getItemPage( ItemId $itemId ): ?Title {
-		return WikibaseFacetedSearchExtension::getInstance()->getItemPageLookup()->getPageTitle( $itemId );
-	}
-
-	private static function newLinkFormatter( Language $language ): EntityLinkFormatter {
-		return WikibaseRepo::getEntityLinkFormatterFactory()->getDefaultLinkFormatter( $language );
-	}
-
-	private static function newLabelDescriptionLookup( IContextSource $context ): LanguageFallbackLabelDescriptionLookup {
-		return new LanguageFallbackLabelDescriptionLookup(
-			WikibaseRepo::getTermLookup(),
-			WikibaseRepo::getLanguageFallbackChainFactory()->newFromContext( $context )
-		);
-	}
-
-	/**
-	 * @param string[] $attributes
-	 */
-	private static function rewriteLinkForNonEntityResult(
-		LanguageFallbackLabelDescriptionLookup $labelDescriptionLookup,
-		EntityLinkFormatter $linkFormatter,
-		ItemId $itemId,
-		string|HtmlArmor|null &$titleSnippet,
-		array &$attributes
-	): void {
-		$labelData = self::termFallbackToTermData(
-			$labelDescriptionLookup->getLabel( $itemId )
-		);
-		$descriptionData = self::termFallbackToTermData(
-			$labelDescriptionLookup->getDescription( $itemId )
-		);
-
-		$titleSnippet = new HtmlArmor( $linkFormatter->getHtml( $itemId, $labelData ) );
-
-		$attributes['title'] = $linkFormatter->getTitleAttribute(
-			$itemId,
-			$labelData,
-			$descriptionData
-		);
-	}
-
-	/**
-	 * @return string[]|null
-	 */
-	private static function termFallbackToTermData( ?TermFallback $term = null ): ?array {
-		if ( $term ) {
-			return [
-				'value' => $term->getText(),
-				'language' => $term->getActualLanguageCode(),
-			];
-		}
-
-		return null;
+		$titleSnippet = $rewrittenLink->titleSnippet ?? $titleSnippet;
+		$attributes = $rewrittenLink->attributes + $attributes;
```

Comment:
I went with a little data class here, instead of mutating the parameters. However, this means some logic is outside of the new service. Thoughts on this versus just mutating in the method?

Reply:
"the new service" = `PageSearchResultRewriter`

"the method" = ?

Reply:
`getRewrittenLink()`

Reply:
Does this question still matter? I cannot find the code in the latest diff